pkg/agent/multicast/mcast_controller.go

```diff
@@ -0,0 +1,362 @@
+	for c.processNextWorkItem() {
+	}
+}
+
+// getGroupMemberStatusByGroup returns the GroupMemberStatus according to the given group.
+func (c *Controller) getGroupMemberStatusByGroup(group net.IP) *GroupMemberStatus {
+	status, ok, _ := c.groupCache.GetByKey(group.String())
+	if ok {
+		return status.(*GroupMemberStatus)
+	}
+	return nil
+}
+
+// getGroupMemberStatusesByPod returns all GroupMemberStatus that the given podInterface is included in its localMembers.
+func (c *Controller) getGroupMemberStatusesByPod(podInterface string) []*GroupMemberStatus {
+	groupMembers := make([]*GroupMemberStatus, 0)
+	statuses, _ := c.groupCache.ByIndex(podInterfaceIndex, podInterface)
+	for _, s := range statuses {
+		groupMembers = append(groupMembers, s.(*GroupMemberStatus))
+	}
+	return groupMembers
+}
+
+func (c *Controller) processNextWorkItem() bool {
+	obj, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	// We expect string (multicast group) to come off the workqueue.
+	if key, ok := obj.(string); !ok {
+		// As the item in the workqueue is actually invalid, we call Forget here else we'd
+		// go into a loop of attempting to process a work item that is invalid.
+		// This should not happen.
+		c.queue.Forget(obj)
+		klog.Errorf("Expected string in work queue but got %#v", obj)
+		return true
+	} else if err := c.syncGroup(key); err == nil {
+		// If no error occurs we Forget this item so it does not get queued again until
+		// another change happens.
+		c.queue.Forget(key)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		c.queue.AddRateLimited(key)
+		klog.Errorf("Error syncing multicast group %s, requeuing. Error: %v", key, err)
+	}
+	return true
+
+}
+
+func (c *Controller) syncGroup(groupKey string) error {
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).Infof("Finished syncing Egress for %s. (%v)", groupKey, time.Since(startTime))
+	}()
+	obj, exists, err := c.groupCache.GetByKey(groupKey)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get GroupMemberStatus", "groupKey", groupKey)
+		return err
+	}
+	if !exists {
+		klog.InfoS("multicast groupKey not found in the cache", "groupKey", groupKey)
+		return nil
+	}
+	status := obj.(*GroupMemberStatus)
+	if c.groupHasInstalled(groupKey) {
+		if c.groupIsStale(status) {
+			// Remove the multicast flow entry if no local Pod has joined the group.
+			if err := c.ofClient.UninstallMulticastFlow(status.group); err != nil {
+				klog.ErrorS(err, "Failed to uninstall multicast flows", "group", groupKey)
+				return err
+			}
+			c.delInstalledGroup(groupKey)
+			status.mutex.Lock()
+			c.groupCache.Delete(status)
+			status.mutex.Unlock()
+			klog.InfoS("Removed multicast group from cache after all members left", "group", groupKey)
+		}
+		return nil
+	}
+	if err := c.ofClient.InstallMulticastFlow(status.group); err != nil {
+		klog.ErrorS(err, "Failed to install multicast flows", "groupKey", status.group)
```

Comment:
The log sometimes use "group", sometimes use "groupKey", can we unify to "group"?